src/test/components/ButtonWithPanel.test.js

```diff
@@ -36,32 +40,30 @@ describe('shared/ButtonWithPanel', function() {
         }
       />
     );
-    expect(button).toMatchSnapshot();
+    expect(container.firstChild).toMatchSnapshot();
   });
 
-  it('renders an opened panel', () => {
-    const button = renderer.create(
+  it('renders a disabled button', () => {
+    const { container } = render(
       <ButtonWithPanel
         className="button"
         label="My Button"
-        open={true}
+        disabled={true}
         panel={
           <ArrowPanel className="panel">
             <div>Panel content</div>
           </ArrowPanel>
         }
       />
     );
-
-    expect(button).toMatchSnapshot();
+    expect(container.firstChild).toMatchSnapshot();
   });
 
-  it('renders a disabled button', () => {
-    const button = renderer.create(
+  it('opens the panel when the button is clicked ', () => {
```

Comment:
nit: you can add at the end: `and closes the panel when the escape key is pressed`